### DISCUSSION TITLE
pg-boss repeats the scheduled completed task after 30 seconds. How to get rid of it?

### DISCUSSION THREAD

**Discussion by galex75** [timgit/pg-boss#473] [Q&A]
pg-boss repeats the scheduled completed task after 30 seconds. How to get rid of it?

**Comment by timgit** [comment]
I'm not sure what your question is.  Please post more details.

**Reply by galex75** [reply]
by running a task on a schedule await boss.schedule(data.name, '*/2 * * * *', null) . it runs after two minutes and after completion it runs again after 30 seconds, and so on indefinitely until you stop it

**Comment by galex75** [comment]
![FireShot Capture 043 - Админка  - localhost](https://github.com/user-attachments/assets/85bba8be-aa30-4eed-b8f5-5b1b59aaf4ab)

**Comment by timgit** [comment]
This should be resolved in 10.0.5

**Comment by galex75** [comment]
I did this and everything worked fine

  `shouldSendIt (cron, tz) {
      const interval = cronParser.parseExpression(cron, { tz })
  
      const prevTime = interval.prev()
  
      const databaseTime = Date.now() + this.clockSkew
  
      const prevDiff = (databaseTime - prevTime.getTime()) / 1000
  
      return prevDiff < 30
    }`

**Comment by timgit** [comment]
I re-opened an issue for this on #480

**Comment by timgit** [comment]
This should be resolved in 10.1.0

**Reply by galex75** [reply]
Thanks, the problem is solved
